cg/services/quality_controller/quality_checks/checks.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable
+from enum import Enum
+
+from cg.constants import Workflow
+from cg.store.models import Case
+from cg.services.quality_controller.quality_checks.utils import (
+    case_pass_sequencing_qc,
+    sample_pass_sequencing_qc,
+    any_sample_in_case_has_reads,
+)
+
+
+class QualityChecks(Enum):
+    """
+    Parent class for the quality checks.
+    """
+
+
+class SequencingQC(QualityChecks):
+    CASE_PASSES: Callable = case_pass_sequencing_qc
+    SAMPLE_PASSES: Callable = sample_pass_sequencing_qc
+    ANY_SAMPLE_IN_CASE_HAS_READS: Callable = any_sample_in_case_has_reads
+
+
+def get_sequencing_quality_check_for_case(case: Case) -> SequencingQC | None:
+    """Get the appropriate sequencing quality checks for the workflow for a case."""
+    workflow: Workflow = case.data_analysis
+
+    case_passes_workflows = [
+        Workflow.BALSAMIC,
+        Workflow.BALSAMIC_PON,
+        Workflow.BALSAMIC_QC,
+        Workflow.BALSAMIC_UMI,
+        Workflow.MIP_DNA,
+        Workflow.MIP_RNA,
+        Workflow.RAREDISEASE,
+        Workflow.RNAFUSION,
+        Workflow.TOMTE,
+    ]
+
+    any_sample_in_case_has_reads_workflows = [
+        Workflow.FLUFFY,
+        Workflow.FASTQ,
+        Workflow.MICROSALT,
+        Workflow.MUTANT,
+        Workflow.TAXPROFILER,
+    ]
+
+    if workflow in case_passes_workflows:
+        return SequencingQC.CASE_PASSES
+    elif workflow in any_sample_in_case_has_reads_workflows:
+        return SequencingQC.ANY_SAMPLE_IN_CASE_HAS_READS
+    else:
+        raise ValueError(f"Workflow {workflow} does not have a sequencing quality check.")
+
+
+def get_sample_sequencing_quality_check() -> SequencingQC:
+    return SequencingQC.SAMPLE_PASSES
+
+
+def run_quality_checks(quality_checks: list[QualityChecks], **kwargs) -> bool:
+    return all(quality_check(**kwargs) for quality_check in quality_checks)
```

Comment:
I would move it to the QCService class:

```suggestion
    @staticmethod
    def _run_quality_checks(quality_checks: list[Callable], **kwargs) -> bool:
        return all(quality_check(**kwargs) for quality_check in quality_checks)
```